packages/sdk/src/chains/BaseAssetTransfer.ts

```diff
@@ -43,4 +58,71 @@ export abstract class BaseAssetTransfer {
 
     return transfer;
   }
+
+  /**
+   * @param {Transfer} transfer Transfer to check
+   * @param {String} destinationProviderUrl URL of the destination chain provider
+   * @returns {Promise<Boolean>} Flag indicating whether there is sufficient balance to execute the transfer on the destination chain
+   */
+  async checkDestinationChainBalance(
```

Comment:
I would remove this function and have separate utility functions for checking liqudity.

Reply:
@mpetrun5 could you elaborate the separate utility functions? Separate for each network type or separate for each token? Where should the Burn/Mint or Lock/Release logic reside?

Reply:
For now I would say per network but see what makes sense.